src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -130,53 +101,34 @@ public ExpenseOD editExpenseInMonth(ExpenseID expenseIDEdit) throws EntityExcept
-                return expenseOB.fail(expenseODFailEdit);
-            }
+                return expenseOB.fail(expenseODFailEdit);}
+
             if(!Objects.equals(expenseIDEdit.getName(), expenseIDEdit.getOldExpense())){
                 if(checkHaveSameNameInList(monthExpenseList, expenseID.getName())) {
                     // Repeated name fail: When a user tries to edit the expense name to a existing expense in month.
                     ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
-                    return expenseOB.fail(expenseODFailEdit);
-                }}
+                    return expenseOB.fail(expenseODFailEdit);}}
+
+            if (changeInRecurringInfo()) {
                 if (expenseID.getIsRecurringExpense()) {
-                    //Repeated recurring expense name fail: When a user tries to edit the current expense to a recurring expense but the name already exist as a recurring expense.
-                    if (checkHaveSameNameInList(recurringExpenseList,expenseID.getName())) {
-                        ExpenseOD expenseODFailEdit = new ExpenseOD("There is a recurring expense with this name, you don't need to add recurring expense in month! " +
-                                "(If this is not the same expense, please use another name!)");
-                        return expenseOB.fail(expenseODFailEdit);
-                    }else {
-                        selectedExpense.setName(expenseIDEdit.getName());
-                        expenseIDEdit.setValue(valueDouble);
-                        selectedExpense.setValue(valueDouble);
-                        session.addRecurExpense(selectedExpense);
-                        selectedExpense.setCategory(selectedCategory);
-                        // Success edit to new recurring expense.
-                        ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this new recurring expense!");
-                        return expenseOB.success(expenseODSuccessEditRecurring);}
-                }else if(checkHaveSameNameInList(recurringExpenseList, expenseID.getOldExpense())&& !expenseIDEdit.getIsRecurringExpense()) {
-                    selectedExpense.setName(expenseIDEdit.getName());
-                    expenseIDEdit.setValue(valueDouble);
-                    selectedExpense.setValue(valueDouble);
+                    setEditInfo(expenseID, valueDouble, selectedExpense,selectedCategory);
+                    session.addRecurExpense(selectedExpense);
+                    // Success edit to new recurring expense.
+                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this new recurring expense!");
+                    return expenseOB.success(expenseODSuccessEditRecurring);
+                }else if (checkHaveSameNameInList(recurringExpenseList, expenseID.getOldExpense())){
```

Comment:
This shouldn't be an else if, it should just be else! Otherwise this block won't run because it has already been failed above when checking for the same name. But other than that minor change this method is good!

Reply:
This else runs when the expense was recurring but now is not! That's why we are deleting it from recurExpense.

Reply:
You're right